### PR TITLE
[FN] Filter coldstaking transactions

### PR DESCRIPTION
**Pull request by zeptin**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs [hunk 1]
```diff
@@ -1420,19 +1420,20 @@ public IActionResult RemoveTransactions([FromQuery]RemoveTransactionsModel reque
                 }
 
                 // If the user chose to resync the wallet after removing transactions.
-                if (result.Any() && request.ReSync)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs:1423]: By removing the `result.Any()` here, are we implying that we will always re-sync regardless if any transactions was affected?

File: src/Stratis.Bitcoin.Features.Wallet/Events/TransactionFound.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using Stratis.Bitcoin.EventBus;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Features.Wallet.Events
+{
+    /// <summary>
+    /// Event that is executed when a transaction is found in the wallet.
+    /// </summary>
+    /// <seealso cref="Stratis.Bitcoin.EventBus.EventBase" />
+    public class TransactionFound : EventBase
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Wallet/Events/TransactionFound.cs:10]: @zeptin I can't seem to see who or what listens for this event? I can see it being published but no listener?

Comment (zeptin) [reply src/Stratis.Bitcoin.Features.Wallet/Events/TransactionFound.cs]: Yeah true, I've removed it

File: src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs (hunk 1 above)
Comment (zeptin) [reply src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs:1423]: Hm, I guess this is debatable. My feeling is that if you've specified a resync, it should happen, even if no changes land up being made.

**Pull request merged**